src/pages/signup.tsx

```diff
@@ -26,10 +35,76 @@ const SignUp = () => {
     createUserWithEmailAndPassword(email, password)
   }
 
+  const onGithubSubmit = async () => {
+    const githubProvider = new firebase.auth.GithubAuthProvider()
+
+    const authUser = await auth
+      .signInWithPopup(githubProvider)
+      .catch(error => {
+        // In the event that an account with this email already exists (because they signed up with email)
+        // Store the existing credential and email conflict, and then ask the user to input the password for their email account
+        // Otherwise... handleErrors()
+        if (error.code === 'auth/account-exists-with-different-credential') {
+          // Store the pending Github credential and conflicting email          
+          setTempCredentials({
+            credential: error.credential,
+            email: error.email,
+            password: null,
+          });
+          // TODO: Open the modal to ask for a password
+          console.log('credential error')   
+        } else {
+          console.log(error)
+        }
+    })
+
+    if (authUser) {
+      // TODO: If we're creating an account for the first time, we need to store some information about the user
+      // see https://github.com/OpenMined/openmined/blob/dev/apps/courses/src/components/forms/users/SignUp.tsx      
+      let firstName = ''
+      let lastName = ''
+
+      if (authUser.user.displayName && authUser.user.displayName !== '') {
+        const splitName = authUser.user.displayName.split(' ');
+
+        firstName = splitName.length >= 1 ? splitName[0] : authUser.user.displayName;
+        lastName = splitName.length >= 2 ? splitName.slice(1).join(' ') : ''
+      }
+
+      const profile = authUser.additionalUserInfo.profile as any
```

Comment:
Avoid using `any` here. The type for the profile can be inferred from the lines below.

Reply:
thanks, I removed the `as any` line